### ISSUE TITLE
508-defect-1 ❗ Launchblocker [SEMANTIC MARKUP]: `<dl>` elements MUST only directly contain properly-ordered `<dt>` and `<dd>` groups, `<script>`, `<template>` or `<div>` elements (Review screen)

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#9867]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

On the Review screen, the code structure must be compliant. There is a description list structure that is missing structural elements. 

### Why it Matters

Screen readers have a specific way of announcing definition lists. When such lists are not properly marked up, this creates the opportunity for confusing or inaccurate screen reader output.

A definition list is used to provide the definitions of words or phrases. The Definition List is marked up using the dl element. Within the list, each term is put in a separate dt element, and its definition goes in the dd element directly following it.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a screen reader user, I want to understand the context of the content.

## Environment

* Operating System: all
* Browser: all
* Screenreading device: all
* Server destination: staging

## Steps to Recreate

1. Enter https://staging.va.gov/hearing-aid-batteries-and-accessories/review-and-submit in browser. logged in as user 1
2. Open each of the accordions
3. Start the axe browser scan
4. Verify that this issue appears

## Possible Fixes (optional)

**Preferred:** Look to see how the review screen was done on Caregiver or Higher Level Review, and use the standard pattern. It's not perfect, but it's consistent.

### Current code
```diff
<dl class="review">
  <div class="address-page">
    <h3 class="vads-u-font-size--h4">Shipping address</h3>
    <dd class="vads-u-margin-top--2">
      <p>We'll ship your order to the address below. Orders typically arrive within 7 to 10 business days.</p>
    </dd>
-    <p class="vads-u-font-weight--bold">Select the address where you'd like to send your order: <span class="vads-u-font-weight--normal schemaform-required-span">(*Required)</span></p>. 
    <div class="review-row">
      <dt>Country</dt>
      <dd><span>United States</span></dd>
    </div>
    <div class="review-row">
      <dt>Street address</dt>
      <dd><span>101 Example Street</span></dd>
    </div>
    <div class="review-row">
      <dt>Line 2</dt>
      <dd><span>Apt 2</span></dd>
    </div>
    <div class="review-row">
      <dt>City</dt>
      <dd><span>Kansas City</span></dd>
    </div>
    <div class="review-row">
      <dt>State</dt>
      <dd><span>Missouri</span></dd>
    </div>
    <div class="review-row">
      <dt>Postal Code</dt>
      <dd><span>64117</span></dd>
    </div>
  </div>
  <div class="review-row">
    <dt>
      <h4>Email address</h4>
      <p>We will send an order confirmation email with a tracking number to this email address.</p>
      <p class="vads-u-margin-bottom--1">Email address</p>
    </dt>
    <dd><span></span></dd>
  </div>
</dl>
```

### Recommended code
```diff
<dl class="review address-page">
    <dt class="vads-u-font-size--h4">Shipping address</dt>
    <dd class="vads-u-margin-top--2">
      <p>We'll ship your order to the address below. Orders typically arrive within 7 to 10 business days.</p>
      <p class="vads-u-font-weight--bold">Select the address where you'd like to send your order: <span class="vads-u-font-weight--normal schemaform-required-span">(*Required)</span></p>. 
    </dd>
    <dd class="review-row">
      <dfn>Country</dfn> <span>United States</span>
    </dd>
    <dd class="review-row">
      <dfn>Street address</dfn> <span>101 Example Street</span>
    </dd>
    <dd class="review-row">
      <dfn>Line 2</dfn> <span>Apt 2</span>
    </dd>
    <dd class="review-row">
      <dfn>City</dfn> <span>Kansas City</span>
    </dd>
    <dd class="review-row">
      <dfn>State</dfn> <span>Missouri</span>
    </dd>
    <dd class="review-row">
      <dfn>Postal Code</dfn> <span>64117</span>
    </dd>

    <dt class="vads-u-font-size--h4">Email address</dt>
    <dd class="review-row">
      <p>We will send an order confirmation email with a tracking number to this email address.</p>
    </dd>
    <dd class="review-row">
      <dfn>Email address</dfn> <span></span>
    </dd>
</dl>
```

## WCAG or Vendor Guidance (optional)

* [`<dl>` elements must only directly contain properly-ordered `<dt>` and `<dd>` groups, `<script>`, or `<template>` elements](https://dequeuniversity.com/rules/axe/3.5/definition-list)

## Screenshots or Trace Logs

<img width="1258" alt="Screen Shot 2020-06-05 at 9 24 41 AM" src="https://user-images.githubusercontent.com/57469/83881463-b538ca00-a70e-11ea-86ea-e5d36b3bc0f9.png">

<img width="728" alt="Screen Shot 2020-06-05 at 9 45 52 AM" src="https://user-images.githubusercontent.com/57469/83883365-5fb1ec80-a711-11ea-9d6f-e63f8843fd3b.png">

**Comment by bradl3yC** [comment]
@jenstrickland can you validate in staging?

**Comment by jenstrickland** [comment]
@bradl3yC The issue is still in staging. To replicate: 

1. On the Review screen, open the Veteran information accordion
1. Run the axe browser scan
1. Verify the issue noted above appears

This would also be caught in the e2e tests, if the scripts open hidden content when they run, which is something that should be in the test scripts.

**Comment by bradl3yC** [comment]
@jenstrickland it looks like this is forms system related - I have changed all the nesting several times and this one looks like it isn't within our code.

**Comment by bradl3yC** [comment]
in the screenshot it looks related to schemaform / accordion component - which is not our code.

**Comment by bradl3yC** [comment]
@jenstrickland this should be ready to validate in staging (part 2)

**Comment by jenstrickland** [comment]
The issue still exists in the Select your supplies review area.

![Screen Shot 2020-06-19 at 11 00 21 AM](https://user-images.githubusercontent.com/57469/85146904-189d1e80-b21c-11ea-8e45-c9615c606878.png)

**Comment by bradl3yC** [comment]
@jenstrickland can you validate in staging? Should be resolved now

**Comment by jenstrickland** [comment]
Validated. 🎉
